kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java

```diff
@@ -241,7 +232,20 @@ public void start(Map<String, String> properties) {
 
   @Override
   public void stop() {
-    logger.trace("task.stop()");
+    try {
+      executor.shutdown();
+      executor.awaitTermination(executor_shutdown_timeout_sec, TimeUnit.SECONDS);
+    } catch (InterruptedException e) {
+      logger.warn("{} active threads are still executing tasks {}s after shutdown is signaled.",
+          executor.getActiveCount(), executor_shutdown_timeout_sec);
+    } finally {
+      logger.trace("task.stop()");
+    }
+  }
+
+  @VisibleForTesting
+  KCBQThreadPoolExecutor getExecutor() {
```

Comment:
This is bugging me. Using dependency injection and taking a factory or lambda in the constructor also bugs me, but at least doesn't introduce a this Google dependency.

Another idea would be to just expose getActiveCount as a public method, and use that in the tests as a proxy for isTerminated.

Reply:
Yeah, I'm not a fan of this either. Making getActiveCount public makes sense to me.

Reply:
Okay, alternatively I could also use java reflection if we want to keep the class clean.